core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -439,18 +545,33 @@ class PartitionTest extends AbstractPartitionTest {
     val leaderEpoch = 5
     val partition = setupPartitionWithMocks(leaderEpoch, isLeader = false)
 
-    def assertReadRecordsError(error: Errors,
-                               currentLeaderEpochOpt: Optional[Integer],
-                               fetchOnlyLeader: Boolean): Unit = {
+    def assertReadRecordsError(
```

Comment:
nit: Is it worth consolidating this one with `assertReadRecordsError` in the previous test? There are identical except that this one takes `clientMetadata` as additional parameter.

Reply:
Thanks. It already looks much better like this.